packages/base/src/panelview/components/legendItem.tsx

```diff
@@ -167,20 +217,89 @@ export const LegendItem: React.FC<{
+                </div>
+
+                {/* Max */}
+                <div
+                  style={{
+                    position: 'absolute',
+                    left: '100%',
+                    transform: 'translateX(-100%)',
+                  }}
+                >
                   <div
                     style={{
-                      position: 'absolute',
-                      top: up ? -18 : 12,
-                      fontSize: '0.7em',
-                      whiteSpace: 'nowrap',
-                      marginTop: up ? 0 : 4,
+                      width: 1,
+                      height: 8,
+                      background: '#333',
+                      margin: '0 auto',
                     }}
+                  />
+                  <div
+                    style={{ position: 'absolute', top: 12, fontSize: '0.7em' }}
                   >
-                    {s.value.toFixed(2)}
+                    {maxValue.toFixed(2)}
                   </div>
                 </div>
-              );
-            })}
+
+                {/* Critical */}
+                {isDivergent && (
+                  <div
+                    style={{
+                      position: 'absolute',
+                      left: `${
+                        criticalValue <= minValue
+                          ? 0
+                          : criticalValue >= maxValue
+                            ? 100
+                            : ((criticalValue - minValue) /
+                                (maxValue - minValue)) *
+                              100
+                      }%`,
+                      transform: 'translateX(-50%)',
+                    }}
+                  >
+                    <div
+                      style={{
+                        width: 2,
+                        height: 14,
+                        background: 'red',
+                        margin: '0 auto',
+                      }}
+                    />
+                    <div
+                      style={{
+                        position: 'absolute',
+                        top: -20,
+                        fontSize: '0.7em',
+                        fontWeight: 'bold',
+                        color: 'red',
+                      }}
+                    >
+                      {criticalValue.toFixed(2)}
+                    </div>
+                  </div>
+                )}
+              </>
+            )}
+          </div>
+
+          {/* Labels below bar */}
+          <div
+            style={{ fontSize: '0.75em', color: '#555', fontWeight: 'bold' }}
+          >
+            {minValue !== undefined && <div>Min: {minValue.toFixed(2)}</div>}
+            {maxValue !== undefined && <div>Max: {maxValue.toFixed(2)}</div>}
+            {isDivergent && criticalValue !== undefined && (
+              <div style={{ color: '#555', fontWeight: 'bold' }}>
+                Critical: {criticalValue.toFixed(2)}
+              </div>
+            )}
```

Comment:
The bar communicates all this information already, I don't think we need the additional labels :)
```suggestion
```